tests/kernel/mem_protect/mpu/src/kobject.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "test_mpu.h"
+
+/* Kernel objects */
+K_THREAD_STACK_DEFINE(kobject_test_4_stack, KOBJECT_STACK_SIZE);
+K_THREAD_STACK_DEFINE(kobject_test_5_1_stack, KOBJECT_STACK_SIZE);
+K_THREAD_STACK_DEFINE(kobject_test_5_2_stack, KOBJECT_STACK_SIZE);
+
+K_SEM_DEFINE(kobject_sem, SEMAPHORE_INIT_COUNT, SEMAPHORE_MAX_COUNT);
+K_MUTEX_DEFINE(kobject_mutex);
+__kernel struct k_thread kobject_test_4_tid;
+__kernel struct k_thread kobject_test_5_1_tid, kobject_test_5_2_tid;
+struct k_sem *random_sem_type=NULL;
```

Comment:
What's the point of this?  If you really want to test what happens when passing NULL, you don't need this variable to do it - you can just pass NULL directly.  The k_thread_access_grant() is just going to treat random_sem_type as the terminating NULL in the list so it isn't truly being "granted".